sonic_data_client/db_client.go

```diff
@@ -465,13 +465,14 @@ func IsTargetDb(target string) (string, bool, string, bool) {
 	dbName := targetname[0]
 	dbNameSpaceExist := false
 	dbNamespace, _ := sdcfg.GetDbDefaultNamespace()
+	isMultiNamespace, _ := sdcfg.CheckDbMultiNamespace()
 
 	if len(targetname) > 2 {
 		log.V(1).Infof("target format is not correct")
 		return dbName, false, dbNamespace, dbNameSpaceExist
 	}
-
-	if len(targetname) > 1 {
+	// ASIC Suffix is only used in case if device is multi-asic/namespace
+	if isMultiNamespace && len(targetname) > 1 {
```

Comment:
"Before the fix client was getting error. After change it is getting data as expected." Can we get UT for this case?